app/models/sdp_code.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+class SdpCode < ApplicationRecord
+  include PgSearch::Model
+
+  pg_search_scope(
+    :search, against: %i[main first second third name], using: { tsearch: { prefix: true } }
+  )
+
+  # rubocop:disable Style/FormatStringToken
+  def full_code
+    tmpl = '%03<main>d.%02<first>d.%02<second>d'
+    format(tmpl, main: main, first: first, second: second)
```

Comment:
`tmpl` ismi yerine doğrudan `template` olarak kullanılması bence daha uygun olur, `tmpl` ilk bakışta biraz belirsiz kalıyor.

Reply:
bu kullan at bir değişken olduğu için çok açık olmaya gerek yok bence. ayrıca `tmpl` bilinen bir değişken adı. go'da çok sık tercih edilir

Reply:
Dar lokal kapsamlarda (bu örnekte iki satır) değişken adları kısa hatta çok kısa (bk. Go ekosistemi) tutulabilir.

P.S. Bana kalsa ara değişken kullanmadan doğrudan `format('...', ...)` olarak da gidilebilir.  Ama bu tür durumlarda önemsizliğine binaen reviewlerde geliştiricinin tercihine bırakıyorum.

Reply:
Madem Go dedik, okuyalım: https://research.swtch.com/names

TLDR;  The descriptiveness of a variable name should be directly related to the size of the scope in which that name can be used.

Reply:
Sıklıkla kullandığım bir diğer ipucu.  Bir "şeyi" değişken yapıp sonrasında uzun ve açıklayıcı bir değişken adı kullanmak durumunda kalmışsanız orada durun ve hemen metod adına yoğunlaşın.  Uzun ve açıklayıcı isimler metod adlarına yakışır.  Örneğin:

```ruby
def calculate
  foo_bar_baz = 'init value'
  format(foo_bar_baz, 3, 5)
end
```

yerine

```ruby
def foo_bar_baz
  format('init value', 3, 5)
end
```

Demem o ki metod adlarında mümkün olduğunca dikkatli ve açıklayıcı olun; zaten yeterince meşakkatli olan "isimlendirme" emeğini metod adlarında harcayın (değişkenler yerine).  Ruby'de yaygın pratik gereği metod uzunluğu zaten 5 satır civarındadır ve bu denli dar bir kapsamda kısa değişken adları nadiren sorun olur.  Metod adları hemen herşeydir.